crates/solidity/outputs/cargo/tests/src/doc_examples/visitor_api.rs

```diff
@@ -17,40 +16,38 @@ struct ContractCollector {
 }
 
 impl Visitor<Error> for ContractCollector {
-    fn enter_rule(
+    fn rule_enter(
         &mut self,
         node: &Rc<RuleNode>,
-        _path: &Vec<Rc<RuleNode>>,
-        _range: &TextRange,
+        _cursor: &Cursor,
     ) -> Result<VisitorEntryResponse> {
-        if node.kind != RuleKind::ContractDefinition {
-            return Ok(VisitorEntryResponse::StepIn);
+        if node.kind == RuleKind::ContractDefinition {
+            if let Node::Token(token) = &node.children[2] {
+                assert_eq!(token.kind, TokenKind::Identifier);
```

Comment:
nit: `ensure!()` to be consistent with `bail!()` instead of panicking?

Reply:
done